runtime/dart_controller.cc

```diff
@@ -95,9 +95,16 @@ bool DartController::SendStartMessage(Dart_Handle root_library) {
 }
 
 tonic::DartErrorHandleType DartController::RunFromKernel(
-    const uint8_t* buffer, size_t size) {
+    const std::vector<uint8_t>& kernel) {
   tonic::DartState::Scope scope(dart_state());
-  Dart_Handle result = Dart_LoadKernel(Dart_ReadKernelBinary(buffer, size));
+  // Copy kernel bytes so they won't go away after we exit this method.
+  // This is needed because original kernel data has to be available
+  // during code execution.
+  uint8_t* kernel_data = (uint8_t*) malloc(kernel.size());
+  memcpy(kernel_data, kernel.data(), kernel.size());
```

Comment:
Fixed. Now I am keeping pointers to those bytes in dart_controller instance and free them when controller is destroyed.